gammacat/dataset.py

```diff
@@ -0,0 +1,47 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import logging
+from .utils import load_yaml, write_yaml
+from gammapy.catalog.gammacat import GammaCatResource
+
+__all__ = [
+    'DataSet',
+]
+
+log = logging.getLogger(__name__)
+
+class DataSet:
+    """Process a dataset file."""
+    resource_type = 'ds'
+
+    def __init__(self, filename):
+        log.debug('DataSet.__init__()')
+        self.filename = filename
+        self.data = load_yaml(filename)
+
+    def write(self, filename):
+        write_yaml(self.data, filename)
+
+    def reference_id(self):
+        return self.data['reference_id']
+
+    def folder(self):
+        return self.data['reference_id'].replace('&', '%26')
+
+    def source_id(self):
+        return self.data['source_id']
+
+    def to_dict(self):
+        return self.data
+
+    def resource(self):
+        try:
```

Comment:
Replace try-except with this line:
```
file_id = self.data.get('file_id', GammaCatResource._NA_FILL['file_id'])
```

Reply:
This does not work, don't know why. Hence, I leave it as it is.

Reply:
Works for me. I'm putting this now. If it doesn't for you, one cause might be that you have an old Gammapy. Also for the future: please post the error or say what / why this doesn't work, i.e. which output file isn't as expected.